### PR TITLE
Create deque_doubly.py

### PR DESCRIPTION
**Pull request by Himan10**
Implementing Deque ADT using Doubly Linked List...
Every operation of Deque using DLL runs in constant time

### CODE REVIEW

**Comment by cclauss** [comment]
No doctests or type hints as discussed in CONTRIBUTING.md

**Comment by Himan10** [comment]
> No doctests or type hints as discussed in CONTRIBUTING.md

Do i need to provide doctests for private class too ?/

**Comment by cclauss** [comment]
> Do i need to provide doctests for private class too?

Only if we want to make sure that it works the way that we think it should.

**Comment by Himan10** [comment]
> > Do i need to provide doctests for private class too?
> 
> Only if we want to make sure that it works the way that we think it should.

okay I'll.. but all the doctests which i put in my code are working on my machine. but why travis CI isn't accepting it?

File: data_structures/linked_list/deque_doubly.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    
+    ### DEqueu Remove Operations (At the front, At the end) ###
+    
+    def remove_first(self):
+        """ removal from the front
+        >>> d = LinkedDeque()
+        >>> d.remove_first()
+        Traceback (most recent call last):
+           ...
+        Exception: List is empty
```
Comment (cclauss) [inline data_structures/linked_list/deque_doubly.py]: This should be the same exception and a similar error message as __[].pop()__ --> IndexError: pop from empty list

So something like __IndexError: remove_first from empty list__

File: data_structures/linked_list/deque_doubly.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        >>> d.add_first('A') # doctest: +ELLIPSIS
+        <deque_doubly.LinkedDeque object at ...
+        >>> d.remove_first()
+        'A'
```
Comment (cclauss) [inline data_structures/linked_list/deque_doubly.py]: Please add a test to make sure that d is empty.

File: data_structures/linked_list/deque_doubly.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            raise Exception('List is empty')
+        return self._delete(self._header._next)
+    
+    def remove_last(self):
+        """ removal in the end
+        >>> d = LinkedDeque()
+        >>> d.remove_last()
+        Traceback (most recent call last):
+           ...
+        Exception: List is empty
```
Comment (cclauss) [inline data_structures/linked_list/deque_doubly.py]: This should be the same exception and a similar error message as __[].pop()__ --> IndexError: pop from empty list

So something like __IndexError: remove_last from empty list__

File: data_structures/linked_list/deque_doubly.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        >>> d.add_first('A') # doctest: +ELLIPSIS
+        <deque_doubly.LinkedDeque object at ...
+        >>> d.remove_last()
+        'A'
```
Comment (cclauss) [inline data_structures/linked_list/deque_doubly.py]: Please add a test to make sure that d is empty.

**Comment by Himan10** [comment]
@cclauss sir, why it's not been approved?

File: data_structures/linked_list/deque_doubly.py (hunk 4 above)
Comment (Himan10) [reply data_structures/linked_list/deque_doubly.py]: Yeah, i've done something like ... 
```
>>> d = LinkedDeque()
>>> d.is_empty()
True
>>> d.add_first('A') # doctest: +ELLIPSIS
<deque_doubly.LinkedDeque object at ...
>>> d.remove_last()
 'A'
>>> d.is_empty()
True
```
is it okay sir?

Comment (cclauss) [reply data_structures/linked_list/deque_doubly.py]:
```
<deque_doubly.LinkedDeque object at ...
—>
<linked_list.deque_doubly.LinkedDeque object at ...
```

Comment (Himan10) [reply data_structures/linked_list/deque_doubly.py]: I don't know why you added those two lines after <deque_doubly ... 
are you running this from the linked_list directory?
